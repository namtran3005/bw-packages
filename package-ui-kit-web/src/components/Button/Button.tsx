import * as React from 'react';

import styled from 'styled-components';

import { ButtonBase as MuiButtonBase } from '@material-ui/core';
import { ButtonBaseProps as MuiButtonBaseProps } from '@material-ui/core/ButtonBase';

import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

import {
  MediaStyledProperty,
  mediaStyledProperties,
} from '../../utils/mediaStyledProperty';

import { Typography } from '../Typography';

import { CircularProgress } from '../CircularProgress';

interface ButtonProps extends MuiButtonBaseProps {
  loading?: boolean;
  variant?: 'hollow' | 'solid';
  background?: ColorsType;
  fullWidth?: MediaStyledProperty<boolean>;
  size?: 'S' | 'M' | 'L' | 'XL';
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  href?: string;
  fontSize?: number;
  selected?: boolean;
  zIndex?: React.CSSProperties['zIndex'];
  target?: string;
  textForeground?: ColorsType;
}

const defaultBackground = 'primaryLilac';
const defaultTextColor = 'text';
const disabledOpacitySolid = 0.3;
const disabledOpacityHollow = 0.25;
const hoverOpacity = 0.75;
const borderWidth = 1.4;

const getButtonBackgroundColor = (props: Partial<ButtonProps>) => {
  return props.selected ? 'text' : props.background || defaultBackground;
};

const getButtonBackground = (props: ButtonProps) => {
  if (props.variant === 'solid') {
    return Colors.get(
      props.selected ? 'text' : getButtonBackgroundColor(props)
    );
  }
  return Colors.getWithAlpha(
    props.selected ? 'text' : 'white',
    props.selected ? 1 : 0
  );
};

const getButtonBorderWidth = (props: ButtonProps) => {
  return props.variant === 'hollow' ? borderWidth : 0;
};

export const getButtonColor = ({ selected }: ButtonProps) => {
  return Colors.get(selected ? 'white' : defaultTextColor);
};

const getButtonBorderColor = (props: ButtonProps) => {
  return props.variant === 'hollow' ? getButtonColor(props) : '';
};

const getButtonPadding = (props: ButtonProps) => {
  const verticalPadding = 0;
  const horizontalPadding =
    props.size === 'S'
      ? 12
      : props.size === 'M'
      ? 16
      : props.size === 'L'
      ? 18
      : 24;
  return `${verticalPadding}px ${horizontalPadding}px`;
};

const getButtonActiveBackground = (props: ButtonProps) => {
  if (props.variant === 'solid') {
    const backgroundColor = props.selected
      ? 'text'
      : props.background || defaultBackground;
    return Colors.getWithAlpha(backgroundColor, props.selected ? 1 : 0.5);
  } else {
    return Colors.get(props.selected ? 'text' : 'backgroundLight');
  }
};

const getButtonOpacity = (props: ButtonProps): number => {
  if (props.loading) {
    return 1;
  }
  if (props.disabled) {
    return props.variant === 'solid'
      ? disabledOpacitySolid
      : disabledOpacityHollow;
  }
  return 1;
};

const getButtonHeight = (props: ButtonProps): number => {
  return props.size === 'S'
    ? 32
    : props.size === 'M'
    ? 40
    : props.size === 'L'
    ? 48
    : 56;
};

const ButtonComponent = styled(MuiButtonBase).withConfig({
  shouldForwardProp: (prop) =>
    [
      'action',
      'buttonRef',
      'className',
      'children',
      'data-test',
      'disabled',
      'disableRipple',
      'disableTouchRipple',
      'focusRipple',
      'focusVisibleClassName',
      'href',
      'id',
      'name',
      'onFocusVisible',
      'onSubmit',
      'onClick',
      'type',
      'TouchRippleProps',
      'value',
      'target',
    ].includes(prop),
})`
  &&& {
    white-space: nowrap;
    border-radius: 32px;
    border: ${getButtonBorderWidth}px solid ${getButtonBorderColor};
    padding: ${getButtonPadding};
    background: ${getButtonBackground};
    color: ${getButtonColor};
    opacity: ${getButtonOpacity};
    display: ${(props) => (props.hidden ? 'none' : 'inline-block')};
    z-index: ${(props) =>
      props.zIndex !== undefined ? props.zIndex : 'initial'};

    ${({ fullWidth }) =>
      mediaStyledProperties({
        width: {
          value: fullWidth,
          transform(value: boolean) {
            return value ? '100%' : 'inherit';
          },
        },
      })}
    .button-text-area {
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      pointer-events: none;
      font-size: ${({ fontSize, size }) =>
        fontSize
          ? `${fontSize}px`
          : size === 'XL'
          ? '1.25rem'
          : size === 'S' || size === 'M'
          ? '0.875rem'
          : '1rem'};
    }

    ${(props) =>
      typeof props.href === 'string'
        ? `height: ${getButtonHeight(props)}px;`
        : `min-height: ${getButtonHeight(props)}px;`}
    &:hover {
      opacity: ${hoverOpacity};
    }

    &:active {
      background: ${getButtonActiveBackground};
    }
  }
`;

const Button: React.FunctionComponent<ButtonProps> = ({
  loading = false,
  disabled = false,
  focusRipple = true,
  variant = 'hollow',
  children,
  disableTouchRipple = true,
  selected,
  size = 'L',
  ...cleanProps
}: ButtonProps) => {
  const isDisabled = disabled || loading;

  return (
    <ButtonComponent
      classes={{
        disabled: 'disabled',
      }}
      {...{
        focusRipple,
        variant,
        size,
        disabled: isDisabled,
        disableTouchRipple,
        selected,
      }}
      {...cleanProps}
    >
      {loading ? (
        <CircularProgress size={26} />
      ) : (
        <div className="button-text-area">
          {size === 'S' ? (
            <Typography.Caption
              textAlign="center"
              {...(selected && { foreground: 'white' })}
            >
              {children}
            </Typography.Caption>
          ) : (
            <Typography.Body
              textAlign="center"
              foreground={cleanProps.textForeground}
            >
              {children}
            </Typography.Body>
          )}
        </div>
      )}
    </ButtonComponent>
  );
};

export { Button, ButtonProps };
