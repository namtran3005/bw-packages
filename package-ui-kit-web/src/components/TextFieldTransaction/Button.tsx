import * as React from 'react';

import styled from 'styled-components';

import { ButtonBase as MuiButtonBase } from '@material-ui/core';
import { ButtonBaseProps as MuiButtonBaseProps } from '@material-ui/core/ButtonBase';

import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

import {
  MediaStyledProperty,
  mediaStyledProperties,
} from '../../utils/mediaStyledProperty';

import { Typography } from '../Typography';

import { CircularProgress } from '../CircularProgress';

interface ButtonProps extends MuiButtonBaseProps {
  loading?: boolean;
  variant?: 'hollow' | 'solid';
  background?: ColorsType;
  fullWidth?: MediaStyledProperty<boolean>;
  size?: 'tiny' | 'small' | 'normal' | 'large';
  fontWeight?: number;
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  href?: string;
  fontSize?: number;
  target?: string;
}

const defaultBackground = 'primaryLilac';
const defaultTextColor = 'primaryBlack';
const disabledOpacitySolid = 0.3;
const disabledOpacityHollow = 0.25;
const hoverOpacity = 0.75;
const borderWidth = 1.4;

const getButtonBackgroundColor = (props: Partial<ButtonProps>) => {
  return props.background || defaultBackground;
};

const getButtonBackground = (props: ButtonProps) => {
  if (props.variant === 'solid') {
    return Colors.get(getButtonBackgroundColor(props));
  }
  return Colors.getWithAlpha('white', 0);
};

const getButtonBorderWidth = (props: ButtonProps) => {
  return props.variant === 'hollow' ? borderWidth : 0;
};

export const getButtonColor = () => {
  return Colors.get(defaultTextColor);
};

const getButtonBorderColor = (props: ButtonProps) => {
  return props.variant === 'hollow' ? getButtonColor : '';
};

const getButtonPadding = (props: ButtonProps) => {
  const verticalPadding = 0;
  const horizontalPadding =
    props.size === 'tiny'
      ? 12
      : props.size === 'small'
      ? 16
      : props.size === 'large'
      ? 24
      : 18;
  return `${verticalPadding}px ${horizontalPadding}px`;
};

const getButtonActiveBackground = (props: ButtonProps) => {
  if (props.variant === 'solid') {
    const backgroundColor = props.background || defaultBackground;
    return Colors.getWithAlpha(backgroundColor, 0.5);
  } else {
    return Colors.get('backgroundLight');
  }
};

const getButtonOpacity = (props: ButtonProps): number => {
  if (props.loading) {
    return 1;
  }
  if (props.disabled) {
    return props.variant === 'solid'
      ? disabledOpacitySolid
      : disabledOpacityHollow;
  }
  return 1;
};

const ButtonComponent = styled(MuiButtonBase).withConfig({
  shouldForwardProp: (prop) =>
    [
      'action',
      'buttonRef',
      'className',
      'children',
      'disabled',
      'disableRipple',
      'disableTouchRipple',
      'focusRipple',
      'focusVisibleClassName',
      'onFocusVisible',
      'type',
      'TouchRippleProps',
      'href',
      'onSubmit',
      'onClick',
    ].includes(prop),
})`
  &&& {
    white-space: nowrap;
    border-radius: 32px;
    border: ${getButtonBorderWidth}px solid ${getButtonBorderColor};
    padding: ${getButtonPadding};
    background: ${getButtonBackground};
    color: ${getButtonColor};
    opacity: ${getButtonOpacity};
    display: ${(props) =>
      props.hidden
        ? 'none'
        : typeof props.href === 'string'
        ? 'table-cell'
        : 'inline-block'};
    ${({ fullWidth }) =>
      mediaStyledProperties({
        width: {
          value: fullWidth,
          transform(value: boolean) {
            return value ? '100%' : 'inherit';
          },
        },
      })}
    .button-text-area > p {
      font-size: 12px !important;
    }
    &:hover {
      opacity: ${hoverOpacity};
    }
    &:active {
      background: ${getButtonActiveBackground};
    }
  }
`;

const Button: React.FunctionComponent<ButtonProps> = ({
  loading = false,
  disabled = false,
  focusRipple = true,
  fontWeight = 500,
  variant = 'hollow',
  children,
  disableTouchRipple = true,
  size = 'normal',
  ...cleanProps
}: ButtonProps) => {
  const isDisabled = disabled || loading;

  return (
    <ButtonComponent
      classes={{
        disabled: 'disabled',
      }}
      {...{
        focusRipple,
        variant,
        size,
        disabled: isDisabled,
        disableTouchRipple,
      }}
      {...cleanProps}
    >
      {loading ? (
        <CircularProgress size={26} />
      ) : (
        <div className="button-text-area">
          {size === 'tiny' ? (
            <Typography.Caption>{children}</Typography.Caption>
          ) : (
            <Typography.Body fontWeight={fontWeight}>
              {children}
            </Typography.Body>
          )}
        </div>
      )}
    </ButtonComponent>
  );
};

export { Button, ButtonProps };
