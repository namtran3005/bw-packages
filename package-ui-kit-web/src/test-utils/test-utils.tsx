import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
