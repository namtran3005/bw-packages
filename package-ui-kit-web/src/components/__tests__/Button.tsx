import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { Button } from '../Button';

const defaultProps = {
  onClick: jest.fn(),
  // eslint-disable-next-line @typescript-eslint/prefer-as-const
  background: 'primaryLilac' as 'primaryLilac',
  target: '_blank',
  rel: 'nofollow',
};

describe('Button', () => {
  it('should render button', () => {
    const { baseElement } = renderWithTestHarness(<Button {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render link', () => {
    const newProps = { ...defaultProps, href: 'url' };
    const { baseElement } = renderWithTestHarness(<Button {...newProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render disabled button', () => {
    const { baseElement } = renderWithTestHarness(<Button disabled />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render solid button', () => {
    const { baseElement } = renderWithTestHarness(<Button variant="solid" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render solid disabled button', () => {
    const { baseElement } = renderWithTestHarness(
      <Button variant="solid" disabled />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render hollow button', () => {
    const { baseElement } = renderWithTestHarness(<Button variant="hollow" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render hollow disabled button', () => {
    const { baseElement } = renderWithTestHarness(
      <Button variant="hollow" disabled />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render small button', () => {
    const { baseElement } = renderWithTestHarness(<Button size="M" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render large button', () => {
    const { baseElement } = renderWithTestHarness(<Button size="XL" />);
    expect(baseElement).toMatchSnapshot();
  });
});
