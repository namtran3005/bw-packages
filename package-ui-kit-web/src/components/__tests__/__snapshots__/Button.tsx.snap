// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
.c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 18px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<body>
  <div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      target="_blank"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Button should render disabled button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 18px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 0.25;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <button
      class="MuiButtonBase-root c0 Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
    </button>
  </div>
</body>
`;

exports[`Button should render hollow button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 18px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Button should render hollow disabled button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 18px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 0.25;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <button
      class="MuiButtonBase-root c0 Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
    </button>
  </div>
</body>
`;

exports[`Button should render large button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 24px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  min-height: 56px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1.25rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Button should render link 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 18px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root c0"
      href="url"
      tabindex="0"
      target="_blank"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</body>
`;

exports[`Button should render small button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 1.4px solid #2C232E;
  padding: 0px 16px;
  background: rgba(255,255,255,0);
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  min-height: 40px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 0.875rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: #F0F0F0;
}

<div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Button should render solid button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 0px solid;
  padding: 0px 18px;
  background: #BEAAFF;
  color: #2C232E;
  opacity: 1;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: rgba(190,170,255,0.5);
}

<div>
    <button
      class="MuiButtonBase-root c0"
      tabindex="0"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Button should render solid disabled button 1`] = `
<body>
  .c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 140%;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c1.c1.c1 {
  text-align: center;
}

.c0.c0.c0 {
  white-space: nowrap;
  border-radius: 32px;
  border: 0px solid;
  padding: 0px 18px;
  background: #BEAAFF;
  color: #2C232E;
  opacity: 0.3;
  display: inline-block;
  z-index: initial;
  min-height: 48px;
}

.c0.c0.c0 .button-text-area {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  font-size: 1rem;
}

.c0.c0.c0:hover {
  opacity: 0.75;
}

.c0.c0.c0:active {
  background: rgba(190,170,255,0.5);
}

<div>
    <button
      class="MuiButtonBase-root c0 Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="button-text-area"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1"
        />
      </div>
    </button>
  </div>
</body>
`;
